draft-ietf-masque-h3-datagram.md

```diff
@@ -303,6 +193,10 @@ Capsule Value:
 : The payload of this capsule. Its semantics are determined by the value of the
 Capsule Type field.
 
+Note that use of the Capsule Protocol is not required to use HTTP Datagrams.
+Definitions of new HTTP Methods or of new HTTP Upgrade Tokens can use HTTP
+Datagrams with their own data stream protocol.
+
 
 ## Requirements
 
```

Comment:
> A server MUST NOT send any Transfer-Encoding or Content-Length header fields in a 2xx (Successful) response. If a client receives a Content-Length or Transfer-Encoding header fields in a successful response, it MUST treat that response as malformed.

If this is just about the response you need to say that:

1. the capsule protocol is not in use for a response unless it includes a 2xx status code
2. the capsule protocol MUST NOT be used with messages that contain Content-Length or Transfer-Encoding

You also say:

> A request message does not have content.

There are cases where Content-Length can be zero, but that is limited, not universal.  It's only CONNECT that gets that special handling.  For a new method, the capsule protocol will need to be used for the request body, which means that the request will have content.

> A successful response message does not have content.

Again, this is not going to be the case for anything other than CONNECT.

> Responses are not cacheable.

Generally true, but not necessarily universally true.  I think that this is more of a recommendation rather than a requirement.  It doesn't make sense to cache a response that uses the capsule protocol, but who are we to say that?

Reply:
Agreed, made the changes